Sources/PayPalNativeCheckout/PayPalClient.swift

```diff
@@ -11,34 +11,52 @@ public class PayPalClient {
 
     public weak var delegate: PayPalDelegate?
 
-    let config: CoreConfig
+    private let apiClient: APIClient
+    private let config: CoreConfig
 
     // swiftlint:disable identifier_name
-    let CheckoutFlow: CheckoutProtocol.Type
+    private let CheckoutFlow: CheckoutProtocol.Type
     // swiftlint:enable identifier_name
 
     /// Initialize a PayPalClient to process PayPal transaction
     /// - Parameters:
     ///   - config: The CoreConfig object
-    ///   - returnURL: The return URL provided to the PayPal Native UI experience. Used as part of the authentication process to identify your application. This value should match the one set in the `Return URLs` section of your application's dashboard on your [PayPal developer account](https://developer.paypal.com)
-    public init(config: CoreConfig) {
-        self.config = config
-        self.CheckoutFlow = Checkout.self
+    public convenience init(config: CoreConfig) {
+        self.init(
+            config: config,
+            checkoutFlow: Checkout.self,
+            apiClient: APIClient(coreConfig: config)
+        )
     }
 
-    init(config: CoreConfig, checkoutFlow: CheckoutProtocol.Type) {
+    init(config: CoreConfig, checkoutFlow: CheckoutProtocol.Type, apiClient: APIClient) {
         self.config = config
         self.CheckoutFlow = checkoutFlow
+        self.apiClient = apiClient
     }
 
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
     ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
     ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) {
-        CheckoutFlow.set(config: config)
+    public func start(request: PayPalRequest, presentingViewController pvc: UIViewController? = nil) async {
```

Comment:
`pvc` feels like a more confusing name for this internal function parameter. Can we just use `presentingViewController`?